apache/superset
# feat(sqllab) Add a configuration option to disable data preview

### SUMMARY
This PR adds a new per-database configuration option to disable data preview queries in SQL Lab. This setting is exposed in the database API as `preview_data`. I decided to add the feature to the database API and pass the database as a parameter to `addTable` rather than adding a database setting into a table API response (the proposed solution in #14726). As a result the data preview query was moved up to `addTable` as well. To my eye that looks cleaner since I don't think that previewing data should be a part of "table metadata" anyway.

There is no UI change, though the UX changes when data preview is disabled to not show the data preview pane (or results) when getting table metadata.

### TESTING INSTRUCTIONS
1. In the database configuration settings, add `"preview_data": false` to the "Extra" JSON object.
2. In SQL Editor, select the database you have configured, and then any schema or table.
    - Alternately, trigger metadata to load from the editor textbox itself.
3. Table metadata should load as normal, but no data preview query should run.
4. When `preview_data` is true (or the key does not exist), a data preview query should run when selecting a table from the dropdown or triggering metadata to be loaded from the editor.
 
### ADDITIONAL INFORMATION


- [x] Has associated issue: Resolves #14726
